### PR TITLE
* add support for android keystores in match

### PR DESCRIPTION
**Pull request by hjanuschka**
Ads support for keeping android keystores and supply p12 files within the match repo.

APK keystores
===

get android key:
```
match(
android: true,
type: "release",
keystore_destination: "../release.keystore"
keystore_alias: "release-key",
keystore_password: "asdf123",
keystore_fullname: "Helmut Januschka",
keystore_city: "Vienna"
)
```

fetches the encrypted keystore from repo, if not found creates a new one, puts the decrypted keystore file to `keystore_destination`

so in our build.gradle we have the following:

```java
signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword "asdf123"
            keyAlias "debug"
            keyPassword "asdf123"
        }
        release {
            storeFile file("release.keystore")
            storePassword "asdf123"
            keyAlias "release"
            keyPassword "asdf123"
        }
    }
```

supply:
===

to get the json file from supply out of my git repo they are in match too now

```
match(
android: true,
supply: true,
type: "release",
supply_destination: "../supply.json"
supply_json_import: "/tmp/original.json"
)
```

fetches and decrypts the supply json certifcate and puts it into: `supply_destination`
optional `supply_json_import` path is used if no supply key already exists and is imported once.
if not set it asks the user for a path.

testing from commandline:

env:

```
export MATCH_KEYSTORE_CITY=Vienna
export MATCH_KEYSTORE_PASSWORD=asdf123
export MATCH_KEYSTORE_DESTINATION=release.keystore
export MATCH_KEYSTORE_FULLNAME=Helmut
export MATCH_KEYSTORE_ALIAS=release
export MATCH_SUPPLY_DESTINATION=/tmp/1.json
export MATCH_SUPPLY_IMPORT_PATH=/Users/h.januschka//fastlane/google_play_key.json
```

keystores:
===

create || get a relase key:

```
./bin/match --android -r git@gitlab.xxx.at:hjanuschka/dd1.git  -y debug
```

create || get a debug key:

```
./bin/match --android -r git@gitlab.xxx.at:hjanuschka/dd1.git  -y debug
```

Nuke it:

```
./bin/match nuke release -r git@gitlab.xx.at:hjanuschka/dd1.git --android
```

supply keys:
===

Get a key or import a new one

```
./bin/match --android -r git@gitlab.xxx.at:hjanuschka/dd1.git  -y debug --supply
```

Nuke it:

```
./bin/match nuke debug -r git@gitlab.xx.at:hjanuschka/dd1.git --android --supply
```

### CODE REVIEW

**Comment by hjanuschka** [comment]
also added `.hound.yml` to use `.rubocop.yml` as in the other fastlane repos - so the hound comments should be not relevant any longer

File: .hound.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ruby:
+  config_file: .rubocop.yml
```
Comment (KrauseFx) [inline .hound.yml]: This should probably be part of a separate PR :+1:

**Comment by KrauseFx** [comment]
First off, thank you so much for this amazing pull request. Moving some further discussion to Slack for now, will update the PR once we're ready :+1:

**Comment by hjanuschka** [comment]
@KrauseFx  got no feedback on slack, anyway

 * moved the hound.yml changes to a own PR
 * rebased master
 * changed to json instead of P12
 * edited my description to use p12 too


looking forward to it.

**Comment by asfalcone** [comment]
Hey @hjanuschka Thanks for this PR. I appreciate the effort you put into this one but I don't think that adding Android key management to the `match` tool is the right strategy so we're not going to merge this PR.

**Comment by hjanuschka** [comment]
@asfalcone  ok, your repo, your rules, its ok for me.
anyway i am still looking for a way to manage my android key/signing stuff using fastlane in a similar way to match (crypted git repo), can you provide a bit more insight about the strategy the team has for this requirement, or is'nt it a topic at'all?
